src/main/scala/io/iohk/ethereum/network/BlockBroadcastActor.scala

```diff
@@ -0,0 +1,173 @@
+import io.iohk.ethereum.network.PeerActor.MessageReceived
+import io.iohk.ethereum.network.PeerManagerActor.{GetPeers, Peer}
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.NewBlock
+import io.iohk.ethereum.network.p2p.messages.PV62
+import io.iohk.ethereum.network.p2p.messages.PV62._
+import io.iohk.ethereum.network.p2p.validators.BlockHeaderError.HeaderParentNotFoundError
+import io.iohk.ethereum.network.p2p.validators.{BlockHeaderValidator, BlockValidator}
+import io.iohk.ethereum.utils.{BlockchainStatus, NodeStatus}
+import org.spongycastle.util.encoders.Hex
+
+//FIXME: Handle known blocks to peers [EC-80]
+class BlockBroadcastActor(
+  nodeStatusHolder: Agent[NodeStatus],
+  peer: ActorRef,
+  peerManagerActor: ActorRef,
+  blockchain: Blockchain) extends Actor with ActorLogging {
+
+  import BlockBroadcastActor._
+
+  override def receive: Receive = idle
+
+  private val msgsToSubscribe = Set(NewBlock.code, PV62.NewBlockHashes.code, BlockHeaders.code, BlockBodies.code)
+
+  def idle: Receive = {
+    case StartBlockBroadcast =>
+      peer ! PeerActor.Subscribe(msgsToSubscribe)
+      context become processMessages(ProcessingState(Seq(), Seq(), Seq(), Seq()))
+  }
+
+  def processMessages(state: ProcessingState): Receive =
+    handleReceivedMessages(state) orElse {
+
+      case ProcessNewBlocks if state.unprocessedBlocks.nonEmpty =>
+        if(state.unprocessedBlocks.tail.nonEmpty) self ! ProcessNewBlocks
+
+        val blockToProcess = state.unprocessedBlocks.head
+        val blockHeader = BlockHeaderValidator.validate(blockToProcess.header, blockchain)
+        val blockParentTotalDifficulty = blockchain.getTotalDifficultyByHash(blockToProcess.header.parentHash)
+
+        (blockHeader, blockParentTotalDifficulty) match {
+          case (Right(_), Some(parentTD)) =>
+            importBlockToBlockchain(blockToProcess, parentTD)
+
+            peerManagerActor ! GetPeers
+            val newState = state.copy(
+              unprocessedBlocks = state.unprocessedBlocks.tail,
+              toBroadcastBlocks = state.toBroadcastBlocks :+ blockToProcess
+            )
+            context become processMessages(newState)
+
+          case (Left(HeaderParentNotFoundError), _) | (Right(_), None) =>
+            log.info("Block parent not found, block {} will not be broadcasted", Hex.toHexString(blockToProcess.header.hash.toArray))
+            context become processMessages(state.copy(unprocessedBlocks = state.unprocessedBlocks.tail))
+
+          case (Left(_), _) =>
+            log.info("Block {} not valid", Hex.toHexString(blockToProcess.header.hash.toArray))
+            context become processMessages(state.copy(unprocessedBlocks = state.unprocessedBlocks.tail))
+        }
+
+      case PeerManagerActor.PeersResponse(peers) if state.toBroadcastBlocks.nonEmpty =>
+        state.toBroadcastBlocks.foreach{ newBlock =>
+          sendNewBlockMsgToPeers(peers, newBlock)
```

Comment:
I thought of incorporating that in a future PR with `Handling of timeouts for the block header and block body requests`. Do you think it would be better to add it here?